### DISCUSSION TITLE
Kill "Vmmem" in task manager

### DISCUSSION THREAD

**Discussion by Msadr471** [microsoft/WSL#6373] [General] [answered]
Hello everyone, how can I close this program that is open in the background! When I open Linux and run the command `kex --sl -s`, a graphical environment opens, but when I exit it, even when I run the command `kex --sl --stop`, this ``Vmmem`` is still open in the background of Windows! Do you have a solution? Occupies a lot of laptop RAM! thank you.
![image](https://user-images.githubusercontent.com/49529241/102995127-9fb5b100-4535-11eb-8821-53a12a8dafc3.png)
![image](https://user-images.githubusercontent.com/49529241/102995763-fbcd0500-4536-11eb-818d-f4ada391b7ed.png)

Kali Linux
![image](https://user-images.githubusercontent.com/49529241/102995702-ddffa000-4536-11eb-8958-242262477b6f.png)

**Comment by therealkenc** [comment, marked as answer]
Variation this [question](https://devblogs.microsoft.com/oldnewthing/20180717-00/?p=99265). Your virtual machine is still running, by design. This is unrelated to 'kex'. You could in principle do `wsl.exe --shutdown`.

**Reply by afilina** [reply]
This doesn't help in cases where `wsl --shutdown` hangs. That's the whole reason for my wanting to force kill that process without a full Windows restart.

**Reply by afilina** [reply]
I'll try this next time, thanks.

**Reply by mr8ug** [reply]
Thank you, that was what i need

**Reply by ivanyv** [reply]
Beware I tried @onomatopellan suggestion and it BSOD on Win11 WSL 0.65.3.0

**Reply by Reonekot** [reply]
To echo ivanyv - killing wmwp.exe instantly BSOD'ed my Windows 10. (Killed it as wmmem was using 100% CPU for hours and wsl shutdown didn't work.)

**Reply by mklueh** [reply]
After using WebStorm with WSL2 and Jetbrains Gateway for the first time, WSL seems to have freezed. First I had to kill Jetbrains Client, but WSL is pretty much dead and even `wsl.exe --shutdown` hangs forever. Any advice?

**Comment by Msadr471** [comment]
Yes, It works `wsl.exe --shutdown`.

![image](https://user-images.githubusercontent.com/49529241/103064363-0fc14700-45c9-11eb-8ce7-6fec5e5c46c4.png)
Why is it still open in the background when I turn off the virtual machine in the background settings? Can't it be closed in the background as soon as the app is closed?

**Reply by craigloewen-msft** [reply]
You may have long running apps that exist inside of the WSL distro. We keep the WSL distro up as long as you have Linux processes running. If you have a service like maybe an ssh daemon this could keep WSL running.
